RenderStatePackager/src/RenderStatePackager.cpp

```diff
@@ -33,217 +33,314 @@ namespace Diligent
 
 RenderStatePackager::RenderStatePackager(RefCntAutoPtr<ISerializationDevice>            pDevice,
                                          RefCntAutoPtr<IShaderSourceInputStreamFactory> pStreamFactory,
+                                         RefCntAutoPtr<IThreadPool>                     pThreadPool,
                                          ARCHIVE_DEVICE_DATA_FLAGS                      DeviceBits) :
     m_pDevice{pDevice},
     m_pStreamFactory{pStreamFactory},
+    m_pThreadPool{pThreadPool},
     m_DeviceBits{DeviceBits}
 {
 }
 
-bool RenderStatePackager::Execute(IArchiver* pArchive, const IRenderStateNotationParser* pNotationParser)
+bool RenderStatePackager::ParseFiles(std::vector<std::string> const& DRSNPaths)
 {
-    DEV_CHECK_ERR(pArchive != nullptr, "pArchive must not be null");
-    DEV_CHECK_ERR(pNotationParser != nullptr, "pNotationParser must not be null");
-    if (pArchive == nullptr || pNotationParser == nullptr)
-        return false;
+    DEV_CHECK_ERR(!DRSNPaths.empty(), "DRSNPaths must not be empty");
 
     try
     {
-        DynamicLinearAllocator Allocator{DefaultRawMemoryAllocator::GetAllocator()};
-
-        auto const& ParserInfo = pNotationParser->GetInfo();
+        std::vector<std::exception_ptr>        Exceptions{};
+        std::vector<RefCntAutoPtr<IAsyncTask>> Tasks{};
```

Comment:
You don't have to keep references to the tasks as thread pool already keeps them. You only need them if you are going to use the tasks, but you are not using them, so this is not necessary.